### PR TITLE
Enable building installers on non-Windows

### PR DESCRIPTION
**Pull request by paulcbetts**
This PR allows you to run grunt-electron-installer to create Windows installers on non-Windows platforms, as long as you have both a recent version of Mono as well as Wine installed.

### CODE REVIEW

File: index.coffee [hunk 1]
```diff
@@ -33,6 +47,7 @@ module.exports = (grunt) ->
 
     {certificateFile, certificatePassword, remoteReleases, signWithParams} = config
 
+    console.log "ASARing!"
```
Comment (kevinsawicki) [inline index.coffee]: :fire:

File: index.coffee [hunk 2]
```diff
@@ -12,9 +12,23 @@ module.exports = (grunt) ->
...
       callback(error)
+      
+  locateExecutableInPath = (name) ->
+    haystack = _.map process.env.PATH.split(/[:;]/), (x) -> path.join(x, name)
+    _.find haystack, (needle) -> fs.existsSync(needle)
 
   grunt.registerMultiTask 'create-windows-installer', 'Create the Windows installer', ->
     @requiresConfig("#{@name}.#{@target}.appDirectory")
+    
+    console.log "Starting!"
```
Comment (kevinsawicki) [inline index.coffee]: :fire:

File: index.coffee [hunk 3]
```diff
@@ -12,9 +12,23 @@ module.exports = (grunt) ->
...
+    
+    useMono = false
+    [monoExe, wineExe] = _.map(['mono', 'wine'], locateExecutableInPath)
+    
+    unless process.platform is 'win32'
+      useMono = true
+      console.log "wine: #{wineExe}, mono: #{monoExe}"
```
Comment (kevinsawicki) [inline index.coffee]: :fire: or should this be logged via the grunt API since it might be useful?

File: index.coffee (hunk 1 above)
Comment (paulcbetts) [inline index.coffee]: Ugh, thought I reset these away, sorry

**Comment by kevinsawicki** [comment]
@paulcbetts Do you think it is worth trying to get the specs running on Mac or Linux CI now using Mono and Wine?

**Comment by kevinsawicki** [comment]
> Do you think it is worth trying to get the specs running on Mac or Linux CI now using Mono and Wine?

Not saying you have to do this as part of this PR, just curious if you've gotten this approach running on Travis already for projects using Squirrel.

**Comment by paulcbetts** [comment]
@kevinsawicki I ran the specs locally and they pass, we could get it up on Travis CI if getting Mono + Wine on the box isn't too much of a PITA

**Comment by kevinsawicki** [comment]
@paulcbetts Looks like the specs are failing on AppVeyor on this line:

```coffee
expect(fs.existsSync(path.join(outputDirectory, 'MyAppSetup.msi'))).toBe true
```

**Comment by kevinsawicki** [comment]
Got the AppVeyor build :green_apple: and also got it :green_apple: on Travis.

Thanks for this @paulcbetts this is so awesome.

**Comment by kevinsawicki** [comment]
Published as `1.1.0` :tada:

**Comment by paulcbetts** [comment]
@kevinsawicki Keep in mind that if you use native node modules you're still up a creek since those can't be built on OS X but we're a step closer at least!

**Comment by kevinsawicki** [comment]
> Keep in mind that if you use native node modules you're still up a creek since

Yeah, perhaps with a steady dose of node-pre-gyp-ing this will be less of an issue going forward.

**Comment by mkaz** [comment]
Great stuff!  I've got the build working on OS X and it even updates the Setup.exe resource correctly, which I'm getting errors when building on Windows. 

Unfortunately, when running the Setup installer on Windows, nothing happens. No install, error or anything. Any suggestions on how to debug? If I do the same build on Windows, besides the resource editing, the installer works.

**Comment by paulcbetts** [comment]
@mkaz What shows up in `C:\Users\USER\AppData\Local\SquirrelTemp`?

**Comment by mkaz** [comment]
The only thing that shows is

`2015-11-12 14:55:07> Program: Starting Squirrel Updater: --install .`

**Comment by paulcbetts** [comment]
Oh! I wonder if we're copying the Mono version of Update.exe into the installer

**Comment by fasterthanlime** [comment]
> Oh! I wonder if we're copying the Mono version of Update.exe into the installer

Yup, that seems to be the case!

updatemono

Any pointers on how to fix it?

**Comment by paulcbetts** [comment]
@fasterthanlime I need to fix it in Squirrel.Window, it's in "Program.cs" if you want to have a try

**Comment by fasterthanlime** [comment]
@paulcbetts I actually got halfway through this file earlier today, but just now found this line, which appears to be the culprit: https://github.com/Squirrel/Squirrel.Windows/blob/master/src/Update/Program.cs#L537

I'm not sure what's the best way to fix it:

  * `join(dirname(assemblyLocation), 'Update.exe')`
  * `assemblyLocation.replace('-Mono.exe', '.exe')`
  * something else entirely?

Pick your favorite & I'll happily submit a Squirrel.Windows PR with it!

**Comment by paulcbetts** [comment]
@fasterthanlime I like that 2nd one

**Pull request merged**